### PR TITLE
Fixes edge case where soft-cliped portion of a read ends up at a nega…

### PR DESCRIPTION
**Pull request by fleharty**
…tive position

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

**Comment by fleharty** [comment]
Made this initial PR, waiting on tests to pass and a larger test on a full sized bam to complete.

**Comment by fleharty** [comment]
@yfarjoun We found an annoying edge case.  Assuming this passes tests, could you take a look at this?

File: src/main/java/picard/sam/AbstractAlignmentMerger.java [hunk 1]
```diff
@@ -819,10 +819,10 @@ private static int getReadPositionAtReferencePositionIgnoreSoftClips(final SAMRe
 
         // Temporarily use the newCigar that has SOFT_CLIPs replaced with MATCH_OR_MISMATCH to get read position at reference, but ignore existence of soft-clips
         rec.setCigar(newCigar);
-        readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos, false);
+        readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos + posShift, false);
```
Comment (yfarjoun) [inline src/main/java/picard/sam/AbstractAlignmentMerger.java:822]: add a comment that since the read effectively got shifted forward by turning the clips into matches, the query position needs also to be moved forward so that it's still querying the same base
